### ISSUE TITLE
GF 6.0 error deploying

### ISSUE THREAD

**Issue by sysmat** [eclipse-ee4j/glassfish#23760]
## Environment Details
* GlassFish Version (and build number): 6.0.0
* JDK version: "1.8.0_112"
* OS: Win 10
* Database: mysql

----------

## Problem Description
- can not deploy war , it work on other app servers as payara, glassfih 4
- stack:
```
[2022-01-07T12:51:09.468+0100] [glassfish 6.0] [SEVERE] [] [jakarta.enterprise.system.tools.deployment.common] [tid: _ThreadID=91 _ThreadName=deployment-jar-scanner] [timeMillis: 1641556269468] [levelValue: 1000] [[
  Exception while visiting jersey/repackaged/com/google/common/collect/AbstractMapBasedMultimap$WrappedSortedSet.class of size 4003
java.lang.IndexOutOfBoundsException: Index: 3, Size: 3
	at java.util.ArrayList.rangeCheck(ArrayList.java:653)
	at java.util.ArrayList.get(ArrayList.java:429)
	at org.glassfish.hk2.classmodel.reflect.impl.MethodModelImpl.getParameter(MethodModelImpl.java:85)
	at org.glassfish.hk2.classmodel.reflect.impl.ModelClassVisitor$ModelMethodVisitor.visitParameterAnnotation(ModelClassVisitor.java:381)
	at org.objectweb.asm.ClassReader.readParameterAnnotations(ClassReader.java:2932)
	at org.objectweb.asm.ClassReader.readMethod(ClassReader.java:1467)
	at org.objectweb.asm.ClassReader.accept(ClassReader.java:717)
	at org.objectweb.asm.ClassReader.accept(ClassReader.java:401)
	at org.glassfish.hk2.classmodel.reflect.Parser$5.on(Parser.java:336)
	at com.sun.enterprise.v3.server.ReadableArchiveScannerAdapter.handleEntry(ReadableArchiveScannerAdapter.java:141)
	at com.sun.enterprise.v3.server.ReadableArchiveScannerAdapter.onSelectedEntries(ReadableArchiveScannerAdapter.java:103)
	at org.glassfish.hk2.classmodel.reflect.Parser.doJob(Parser.java:321)
	at org.glassfish.hk2.classmodel.reflect.Parser.access$300(Parser.java:44)
	at org.glassfish.hk2.classmodel.reflect.Parser$3.call(Parser.java:280)
	at org.glassfish.hk2.classmodel.reflect.Parser$3.call(Parser.java:269)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
]]

[2022-01-07T12:51:10.011+0100] [glassfish 6.0] [INFO] [] [jakarta.enterprise.system.tools.deployment.common] [tid: _ThreadID=48 _ThreadName=admin-listener(1)] [timeMillis: 1641556270011] [levelValue: 800] [[
  visiting unvisited references]]

[2022-01-07T12:51:10.799+0100] [glassfish 6.0] [SEVERE] [] [jakarta.enterprise.system.tools.deployment.dol] [tid: _ThreadID=48 _ThreadName=admin-listener(1)] [timeMillis: 1641556270799] [levelValue: 1000] [[
  Requested schema is not found in local repository, please ensure that there are no typos in the XML namespace declaration.
org.xml.sax.SAXException: Requested schema is not found in local repository, please ensure that there are no typos in the XML namespace declaration.
	at com.sun.enterprise.deployment.node.SaxParserHandler.resolveEntity(SaxParserHandler.java:229)
	at com.sun.org.apache.xerces.internal.util.EntityResolverWrapper.resolveEntity(EntityResolverWrapper.java:110)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.resolveEntity(XMLEntityManager.java:1081)
	at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaLoader.resolveDocument(XMLSchemaLoader.java:657)
	at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaValidator.findSchemaGrammar(XMLSchemaValidator.java:2430)
	at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaValidator.handleStartElement(XMLSchemaValidator.java:1767)
	at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaValidator.startElement(XMLSchemaValidator.java:740)
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.scanStartElement(XMLNSDocumentScannerImpl.java:374)
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl$NSContentDriver.scanRootElementHook(XMLNSDocumentScannerImpl.java:613)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(XMLDocumentFragmentScannerImpl.java:3132)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$PrologDriver.next(XMLDocumentScannerImpl.java:852)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:602)
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:112)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:505)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:841)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:770)
	at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
	at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1213)
	at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:643)
	at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl.parse(SAXParserImpl.java:327)
	at com.sun.enterprise.deployment.io.DeploymentDescriptorFile.read(DeploymentDescriptorFile.java:294)
	at com.sun.enterprise.deployment.archivist.ExtensionsArchivist.open(ExtensionsArchivist.java:155)
	at com.sun.enterprise.deployment.archivist.PersistenceArchivist.readPersistenceDeploymentDescriptor(PersistenceArchivist.java:120)
	at com.sun.enterprise.deployment.archivist.WarPersistenceArchivist.open(WarPersistenceArchivist.java:69)
	at com.sun.enterprise.deployment.archivist.Archivist.readRestDeploymentDescriptors(Archivist.java:381)
	at com.sun.enterprise.deployment.archivist.Archivist.readDeploymentDescriptors(Archivist.java:372)
	at com.sun.enterprise.deployment.archivist.Archivist.open(Archivist.java:247)
	at com.sun.enterprise.deployment.archivist.Archivist.open(Archivist.java:256)
	at com.sun.enterprise.deployment.archivist.Archivist.open(Archivist.java:217)
	at com.sun.enterprise.deployment.archivist.ApplicationFactory.openArchive(ApplicationFactory.java:137)
	at org.glassfish.javaee.core.deployment.DolProvider.processDOL(DolProvider.java:182)
	at org.glassfish.javaee.core.deployment.DolProvider.load(DolProvider.java:206)
	at org.glassfish.javaee.core.deployment.DolProvider.load(DolProvider.java:72)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.loadDeployer(ApplicationLifecycle.java:857)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.setupContainerInfos(ApplicationLifecycle.java:797)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:354)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:195)
	at org.glassfish.deployment.admin.DeployCommand.execute(DeployCommand.java:467)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:516)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:512)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:360)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:511)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:542)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:534)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:360)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:533)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:1441)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.access$1300(CommandRunnerImpl.java:86)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1823)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1699)
	at com.sun.enterprise.v3.admin.AdminAdapter.doCommand(AdminAdapter.java:510)
	at com.sun.enterprise.v3.admin.AdminAdapter.onMissingResource(AdminAdapter.java:200)
	at org.glassfish.grizzly.http.server.StaticHttpHandlerBase.service(StaticHttpHandlerBase.java:150)
	at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:440)
	at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:144)
	at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:174)
	at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:153)
	at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:196)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:88)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:246)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:178)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:118)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:96)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:51)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:510)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:82)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:83)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:34)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:101)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:535)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:515)
	at java.lang.Thread.run(Thread.java:745)
]]

```

## Impact of Issue
- run web application on app server
- logging of error should be more descriptive, wich schema can not find

**Comment by sysmat** [comment]
in all descriptors, schema locations are as http links are valid and reachable by net

**Comment by dmatej** [comment]
Hi, I believe nobody will fix it now in GF 6.0 (except you or someone else from the community), could you try 6.2.4 or 7.0.0-M1? At least if the result is the same.
I will take a look on the logging soon.

**Comment by dmatej** [comment]
There's a new issue for the logging: https://github.com/eclipse-ee4j/glassfish/issues/23601
Those descriptors - I would guess that http is prohibited now and everything must be used with https. And that was fixed this week: https://github.com/eclipse-ee4j/glassfish/pull/24062/files 

Can you reproduce it at least on GlassFish 6.2.5 ? 6.0.0 is really obsoleted now.
